src/peer_manager.rs

```diff
@@ -316,6 +320,9 @@ impl PeerManager {
             let state = PeerState::Routing(tunnel);
             let _ = self.peer_map.insert(Peer::new(pub_id, Some(peer_id), state));
         }
+
+        let _ = self.unknown_peers.remove(&peer_id);
```

Comment:
I'm not sure if this is the only place where we should try to remove the peer from the `unknown_peers` map.